src/core/mac/mac.cpp

```diff
@@ -96,16 +97,29 @@ void Mac::StartCsmaBackoff(void)
             backoffExponent = kMaxBE;
         }
 
-        backoff = kMinBackoff + (kUnitBackoffPeriod * kPhyUsPerSymbol * (1 << backoffExponent)) / 1000;
-        backoff = (otPlatRandomGet() % backoff);
+        backoff = (otPlatRandomGet() % (1UL << backoffExponent));
```

Comment:
For CPU efficiency this could be;

backoff = (otPlatRandomGet() & ((1UL << backoffExponent) - 1));